azkaban-common/src/test/java/azkaban/user/XmlUserManagerTest.java

```diff
@@ -73,6 +78,59 @@ public void testDoNotExist() throws Exception {
     fail("XmlUserManager should throw an exception when the file doesn't exist");
   }
 
+  /**
+   * Test auto reload of user XML
+   */
+  @Test
+  public void testAutoReload() throws Exception {
+    final Props props = new Props(this.baseProps);
+    final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
+    final String path = configURL.getPath();
+    props.put(XmlUserManager.XML_FILE_PARAM, path);
+
+    UserManager manager;
+    try {
+      manager = new XmlUserManager(props);
+    } catch (final RuntimeException e) {
+      fail("Should have found the xml file");
+      return;
+    }
+
+    // Get the user8 from existing XML with password == password8
+    User user8 = manager.getUser("user8", "password8");
+
+    // Modify the password for user8
+    // TODO : djaiswal : Find a better way to modify XML
+    List<String> lines = new ArrayList<>();
+    List<String> origLines = new ArrayList<>();
+    for (String line : Files.readAllLines(Paths.get(path))) {
+      origLines.add(line);
+      if (line.contains("password8")) {
+        lines.add(line.replace("password8", "passwordModified"));
+      } else {
+        lines.add(line);
+      }
+    }
+    // Update the file
+    Files.write(Paths.get(path), lines);
+
+    // This should reload the XML file. Check for user8 again
+    try {
+      user8 = manager.getUser("user8", "password8");
```

Comment:
It's not clear to me what the test is asserting. Could you elaborate?

Reply:
The config file originally contains user/password "user8" and "password8". The password is modified in "files.write() call which results in a watch event which reloads the config.
This try catch block fetches a user with the old password but should fail with "does not exist error" which is caught in the catch block.
In the subsequent try catch block, if an exception occurs then the test failed as it uses the updated password to fetch the user.

Reply:
I see. Would it be more straightforward to test the value?

Reply:
Edwina also suggested something similar. Will update the patch.